test-unit/unit/record/json-pathSpec.js

```diff
@@ -137,6 +137,16 @@ describe( 'objects are created from paths and their value is set correctly', fun
 		});
 	});
 
+	it( 'even when the path is not NaNish and could be interpreted as a base 16 number', function() {
+		var record = {};
+		let pathName = '0x02335';
```

Comment:
let fails build ( we can put a use strict at the top since this just tests though if you want )